### PR TITLE
B:18520 Update Nav Links To Return To Move Page (Integration)(MERGE AFTER 18654)

### PR DESCRIPTION
**Pull request by TevinAdams**
## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18520)

MERGE AFTER 18654 
## Summary

Went through the milmove multimove workflows and updated the page navigations to route to customerRoutes.MOVE_HOME_PAGE (The Move) where needed. Let me know if I missed any workflows or navigations.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the milmove app as a service member
2. Go through the workflow of creating a user then a move. Go through the workflow of creating shipments.
3. Go through all workflows to catch where anywhere in the workflows navigations it should route back to the move.
4. Go through the edit workflows and backing out of the edit workflows.
5. Go through the workflows saving changes.

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

**Review by traskowskycaci — COMMENTED** [review]
After I submit the move as the customer, which page should I see next? Currently I am routed to the dashboard rather than the move page itself.

**Review by traskowskycaci — CHANGES_REQUESTED** [review]
Seeing the Oops, something went wrong page when I:
- Go through customer flow
- Whenever I can (either before or after submitting move) go to the Edit Profile page.
- Click the Edit next to Contact Info
image
- Click the save button at the bottom of the page. I didn't change any values before saving
- Click Return to move, which tries to route me to http://milmovelocal:3000/move/:moveId. This is where I see the something went wrong page. Seemed related to orders.
image
CCing @danieljordan-caci to see if this popped up in any of the previous PRs on orders

**Comment by TevinAdams** [comment]
> After I submit the move as the customer, which page should I see next? Currently I am routed to the dashboard rather than the move page itself.

This is fixed. Missed the Agreement nav link.

File: src/constants/routes.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ export const customerRoutes = {
   MOVE_AGREEMENT_PATH: '/moves/:moveId/agreement',
   EDIT_PROFILE_PATH: '/moves/review/edit-profile',
   EDIT_OKTA_PROFILE_PATH: '/moves/review/edit-okta-profile',
-  PROFILE_PATH: '/service-member/profile',
+  PROFILE_PATH: '/service-member/profile/:moveId',
```
Comment (danieljordan-caci) [inline src/constants/routes.js:43]: I would avoid using the `:moveId` here. I would say we should either just navigate the user `HOME_PATH` path when hitting save or use `state` to store the `moveId` when navigating around the profile screen. Similar to how the `needsToVerifyProfile` value is passed around.

**Review by danieljordan-caci — DISMISSED** [review]

**Comment by danieljordan-caci** [comment]
> Seeing the Oops, something went wrong page when I:
> 
> * Go through customer flow
> * Whenever I can (either before or after submitting move) go to the Edit Profile page.
> * Click the Edit next to Contact Info
>   image
> * Click the save button at the bottom of the page. I didn't change any values before saving
> * Click Return to move, which tries to route me to http://milmovelocal:3000/move/:moveId. This is where I see the something went wrong page. Seemed related to orders.
>   image
>   CCing @danieljordan-caci to see if this popped up in any of the previous PRs on orders

It's an issue with how he's using the params. I added a comment.

**Review by cameroncaci — DISMISSED** [review]
On new customer -> set up profile -> reach multi move dashboard -> click profile icon -> click return to move -> error page

Additionally, after reaching this error page navigating back to profile appears to not work. Should be resolved by resolving the former

image

image


image

**Comment by TevinAdams** [comment]
> Seeing the Oops, something went wrong page when I:
> 
> * Go through customer flow
> * Whenever I can (either before or after submitting move) go to the Edit Profile page.
> * Click the Edit next to Contact Info
>   image
> * Click the save button at the bottom of the page. I didn't change any values before saving
> * Click Return to move, which tries to route me to http://milmovelocal:3000/move/:moveId. This is where I see the something went wrong page. Seemed related to orders.
>   image
>   CCing @danieljordan-caci to see if this popped up in any of the previous PRs on orders

This is fixed.

**Comment by TevinAdams** [comment]
> After I submit the move as the customer, which page should I see next? Currently I am routed to the dashboard rather than the move page itself.

This has been changed to send you back to the move.

File: src/constants/routes.js (hunk 1 above)
Comment (TevinAdams) [reply src/constants/routes.js]: This change no longer exist. Found a better way to do this.

**Review by traskowskycaci — COMMENTED** [review]
Overall flowed nicely! I saw all the previous issues fixed, looking good.

I did see the Oops something went wrong page again when I:
- submitted the move
- Clicked Review your request
- Clicked the edit on the Edit next to the Profile on this page:
image

**Comment by traskowskycaci** [comment]
> Overall flowed nicely! I saw all the previous issues fixed, looking good.
> 
> I did see the Oops something went wrong page again when I:
> 
> * submitted the move
> * Clicked Review your request
> * Clicked the edit on the Edit next the Profile on this page:
>   image

The page I see that on is: http://milmovelocal:3000/service-member/profile

**Comment by danieljordan-caci** [comment]
> > Overall flowed nicely! I saw all the previous issues fixed, looking good.
> > I did see the Oops something went wrong page again when I:
> > 
> > * submitted the move
> > * Clicked Review your request
> > * Clicked the edit on the Edit next to the Profile on this page:
> >   image
> 
> The page I see that on is: http://milmovelocal:3000/service-member/profile

Do you see any errors in console?

**Comment by TevinAdams** [comment]
> > > Overall flowed nicely! I saw all the previous issues fixed, looking good.
> > > I did see the Oops something went wrong page again when I:
> > > 
> > > * submitted the move
> > > * Clicked Review your request
> > > * Clicked the edit on the Edit next to the Profile on this page:
> > >   image
> > 
> > 
> > The page I see that on is: http://milmovelocal:3000/service-member/profile
> 
> Do you see any errors in console?

This has been fixed. Needed to add moveId to state for when needing to navigate to the Profile page.

**Comment by TevinAdams** [comment]
Actively fixing this for creating a move

File: config/env/stg.app.env [hunk 2]
```diff
@@ -33,5 +33,5 @@ TELEMETRY_ENDPOINT=localhost:4317
 TELEMETRY_USE_XRAY_ID=true
 TLS_ENABLED=true
 SERVE_PRIME_SIMULATOR=true
-FEATURE_FLAG_MULTI_MOVE=false
-FEATURE_FLAG_COUNSELOR_MOVE_CREATE =false
\ No newline at end of file
+FEATURE_FLAG_MULTI_MOVE=true
```
Comment (danieljordan-caci) [inline config/env/stg.app.env:36]: We don't want this to be turned on in staging, right? @deandreJones

File: src/pages/MyMove/Home/MoveHome.jsx [hunk 3]
```diff
@@ -76,7 +76,8 @@ const MoveHome = ({ serviceMemberMoves, isProfileComplete, serviceMember, signed
   // loading the moveId in params to select move details from serviceMemberMoves in state
   const { moveId } = useParams();
   const navigate = useNavigate();
-
+  let { state } = useLocation();
+  state = { ...state, moveId };
```
Comment (danieljordan-caci) [inline src/pages/MyMove/Home/MoveHome.jsx:80]: I don't think you need this here, do you? You can just use `state` as is and keep it `const`, right?
`const { state } = useLocation();`

And then use it like you do on line 321

Comment (danieljordan-caci) [reply src/pages/MyMove/Home/MoveHome.jsx:80]: Plus, `moveId` is already declared above in `useParams()`

Comment (TevinAdams) [reply src/pages/MyMove/Home/MoveHome.jsx:80]: Yeah beacuse I am setting moveId in state to allow navigation back to this page using that move id.

File: src/pages/MyMove/Profile/Profile.jsx [hunk 4]
```diff
@@ -37,16 +39,21 @@ const Profile = ({ serviceMember, currentOrders, currentBackupContacts, moveIsIn
   };
   const [needsToVerifyProfile, setNeedsToVerifyProfile] = useState(false);
   const [profileValidated, setProfileValidated] = useState(false);
+  const [multiMove, setMultiMove] = useState(false);
 
   const navigate = useNavigate();
   const { state } = useLocation();
+  const { moveId } = state;
```
Comment (danieljordan-caci) [inline src/pages/MyMove/Profile/Profile.jsx:46]: Getting the Oops page when navigating to the user's profile from outside a move or link (just clicking the profile icon in the nav bar).

You'll need to add conditional logic to handle null values if `moveId` is not in `state`

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]

File: src/pages/MyMove/Home/MoveHome.jsx (hunk 3 above)
Comment (danieljordan-caci) [reply src/pages/MyMove/Home/MoveHome.jsx:80]: Gotcha

File: src/pages/MyMove/Profile/Profile.jsx (hunk 4 above)
Comment (danieljordan-caci) [reply src/pages/MyMove/Profile/Profile.jsx:46]: And you'll need to do that for every page you can access from outside of the `MoveHome` component

Comment (TevinAdams) [reply src/pages/MyMove/Profile/Profile.jsx:46]: Honestly I just did the work to store the moveId in state. So much easier and less of a headache. This most likely had to give access in a clean and proper way to the moveId.

Comment (TevinAdams) [reply src/pages/MyMove/Profile/Profile.jsx:46]: Code pushed up and this is fixed.

File: src/pages/MyMove/Multi-Moves/MultiMovesMoveContainer/MultiMovesMoveContainer.jsx [hunk 5]
```diff
@@ -9,13 +10,16 @@ import ButtonDropdownMenu from '../../../../components/ButtonDropdownMenu/Button
 
 import styles from './MultiMovesMoveContainer.module.scss';
 
+// import { SET_MOVE_ID } from 'store/general/actions';
```
[the remark this answers is not in the document]
Comment (TevinAdams) [reply src/pages/MyMove/Multi-Moves/MultiMovesMoveContainer/MultiMovesMoveContainer.jsx]: Done.

**Comment by danieljordan-caci** [comment]
When clicking the `Edit` button to edit the Okta profile, it takes me to the profile edit page and not the okta edit page.

https://github.com/transcom/mymove/assets/136510600/ac6c2cb3-e7ff-4646-9f95-66122d71480e

**Comment by danieljordan-caci** [comment]
> When clicking the `Edit` button to edit the Okta profile, it takes me to the profile edit page and not the okta edit page.
> 
>  Screen.Recording.2024-03-14.at.10.26.10.AM.mov

Once this is fixed, I'll stamp ya

**Comment by TevinAdams** [comment]
> When clicking the `Edit` button to edit the Okta profile, it takes me to the profile edit page and not the okta edit page.
> 
>  Screen.Recording.2024-03-14.at.10.26.10.AM.mov

Changed this back to ` editURL={customerRoutes.EDIT_OKTA_PROFILE_PATH}`. Over looked it routing to the Contact Info Path. @danieljordan-caci  Fixed.

File: config/env/stg.app.env (hunk 2 above)
Comment (deandreJones) [reply config/env/stg.app.env:36]: off by default and utilize FF repo to control

File: src/pages/MyMove/Profile/Profile.jsx [hunk 6]
```diff
@@ -37,16 +39,21 @@ const Profile = ({ serviceMember, currentOrders, currentBackupContacts, moveIsIn
...
 
   useEffect(() => {
     if (state && state.needsToVerifyProfile) {
       setNeedsToVerifyProfile(state.needsToVerifyProfile);
     } else {
       setNeedsToVerifyProfile(false);
     }
+    isMultiMoveEnabled().then((enabled) => {
```
Comment (deandreJones) [inline src/pages/MyMove/Profile/Profile.jsx:54]: https://github.com/transcom/mymove/blob/214ae2d7387493d4b73c633312a37fd4d414ebc0/src/utils/featureFlags.js#L137

need to merge in integration, and make adjustments to how FF check it made

Comment (TevinAdams) [reply src/pages/MyMove/Profile/Profile.jsx]: Replaced `isMultiMoveEnabled().then()` with `isBooleanFlagEnabled('multi_move').then()`

Comment (TevinAdams) [reply src/pages/MyMove/Profile/Profile.jsx]: @deandreJones  Done

**Comment by traskowskycaci** [comment]
> This has been fixed. Needed to add moveId to state for when needing to navigate to the Profile page.

Looks good! 

I did notice that from the 
"Review your request" button 
-> "Review your details" page 
-> clicking that edit next to the profile 
-> going to the Profile page, I can only "Return to Move" from the button at the top of the page 
-> end back at the move page

Alternatively, If I click the  "Review your request" button 
-> "Review your details" page 
-> clicking that edit next to the Orders 
-> scrolling to the bottom of the page, I click cancel 
-> end back at the "Review your details page"

Just a difference in routes I noticed. Honestly could go either way on it. Anyone have thoughts?

**Comment by danieljordan-caci** [comment]
> > This has been fixed. Needed to add moveId to state for when needing to navigate to the Profile page.
> 
> Looks good!
> 
> I did notice that from the "Review your request" button -> "Review your details" page -> clicking that edit next to the profile -> going to the Profile page, I can only "Return to Move" from the button at the top of the page -> end back at the move page
> 
> Alternatively, If I click the "Review your request" button -> "Review your details" page -> clicking that edit next to the Orders -> scrolling to the bottom of the page, I click cancel -> end back at the "Review your details page"
> 
> Just a difference in routes I noticed. Honestly could go either way on it. Anyone have thoughts?

I say that's working as intended. Don't mind it.

**Comment by traskowskycaci** [comment]
> I say that's working as intended. Don't mind it.

Won't argue with that. Looks good @TevinAdams!

**Review by traskowskycaci — APPROVED** [review]

**Review by danieljordan-caci — APPROVED** [review]

**Pull request merged**